### PR TITLE
feat(question-type-ranking): #FOR-540 drag&drop ranking's type responses

### PR DESCRIPTION
**Pull request by c4role**
## Describe your changes
Add drag&drop in respond view for question type ranking

## Checklist tests
When you respond to a question type ranking, try to drag&drop your answer

## Issue ticket number and link
[FOR-540](https://jira.support-ent.fr/browse/FOR-540)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: formulaire/src/main/resources/public/ts/directives/question/respond-question-item.ts [hunk 1]
```diff
@@ -280,6 +283,30 @@ export const respondQuestionItem: Directive = ng.directive('respondQuestionItem'
...
+                    for (let i = 0; i < respDrag.length; i++) {
+                        Sortable.create(respDrag[i], {
+                            group: 'drag-container',
+                            animation: 150,
+                            fallbackOnBody: true,
+                            swapThreshold: 0.65,
+                            ghostClass: "sortable-ghost",
+                            onEnd: async function (evt) {
+                                let refresh = await RankingUtils.onEndRankingDragAndDrop(evt, vm.responses);
+                                if (refresh) {
```
Comment (Rseuret) [inline formulaire/src/main/resources/public/ts/directives/question/respond-question-item.ts:300]: Refresh est tout le temps faux non ?

Comment (c4role) [reply formulaire/src/main/resources/public/ts/directives/question/respond-question-item.ts:300]: En effet tu as raison

File: common/src/main/resources/ts/utils/ranking.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import {angular} from "entcore";
+
+export class RankingUtils {
+    static onEndRankingDragAndDrop = (evt: any, responses: Responses): boolean => {
+        let elem = evt.item.firstElementChild.firstElementChild;
+        let scopElem = angular.element(elem).scope().vm;
+        let itemId = scopElem.question.id;
+        let item: Response;
+
+        item = (responses as Responses).all.filter(q => q.question_id === itemId)[0] as Response;
```
Comment (qperie) [inline common/src/main/resources/ts/utils/ranking.ts:11]: Tu peux faire une seule ligne ici

**Review by qperie — APPROVED** [review]
Top d'avoir ajouté des typages au passage aussi ! 👍

**Review by Rseuret — APPROVED** [review]

**Pull request merged**